hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java

```diff
@@ -105,4 +106,27 @@ public String askQuestion(Question question)
             throw new HxInsightConnectorRuntimeException("Failed to ask question", e);
         }
     }
+
+    public AnswerResponse getAnswer(String questionId)
+    {
+        try
+        {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(config.getAnswerUrl().formatted(questionId)))
+                    .headers(authService.getAuthHeaders())
+                    .GET()
+                    .build();
+
+            HttpResponse<String> httpResponse = client.send(request, BodyHandlers.ofString());
+            log.atDebug().log("Question with id {} received a following answer {}", questionId, httpResponse.body());
+
+            throwExceptionOnUnexpectedStatusCode(httpResponse.statusCode(), SC_OK);
```

Comment:
What will be the final server's response status code for thrown let say `EndpointClientErrorException`? `500` or `httpResponse.statusCode()`?

Reply:
we'll have to chage this part anyway to something like this:
<img width="708" alt="image" src="https://github.com/Alfresco/hxinsight-connector/assets/62990104/26b09349-d809-46ee-8a4e-ed67fd3f576f">

Reply:
Will do

Reply:
As discussed - that will be done in a separate PR globally.